src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -84,7 +86,13 @@ private DelegatedExecution getDelegatedExecutionForFieldDefinition(GraphQLFieldD
                                                                 List<MergedField> mergedFields,
                                                                 DelegatedExecution delegatedExecution,
                                                                 ExecutionStepInfo executionStepInfo) {
-        Document query = mergedFieldsToDocument.mergedSelectionSetToDocument(mergedFields);
+        SourceQueryTransformer queryTransformer = new SourceQueryTransformer(context);
+        List<Field> fields = mergedFields.stream()
+                .map(MergedField::getSingleField)
```

Comment:
we need to transform all fields inside the merged fields

Reply:
can you explain more? I thought that merged fields are the same fields repeated multiple times? So for multiple fields inside merged field I would get singular result ? Is this not how it works?

Reply:
Ahh okay, I just read the javadoc for MergedFields.
So I just need transform it one by one (or is there a way of getting merged subselections for merged field) ?

Reply:
this is the one thing missing before we should merge it:

we should have a test case for queries like:

```graphql
{ foo { id} foo{ name}} 
```
this will fail atm because we don't transform all fields in the mergedField